compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5375,9 +5375,8 @@ private void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr) {
             actualType = BUnionType.create(null, new LinkedHashSet<>(nonErrorTypes));
         }
 
-        if (actualType.tag == TypeTags.NEVER) {
-            dlog.error(checkedExpr.pos, DiagnosticErrorCode.NEVER_TYPE_NOT_ALLOWED_WITH_CHECKED_EXPR,
-                    operatorType);
+        if (types.isNeverTypeOrStructureTypeWithARequiredNeverMember(actualType)) {
+            dlog.error(checkedExpr.expr.pos, DiagnosticErrorCode.EXPRESSION_OF_NEVER_TYPE_NOT_ALLOWED);
         }
 
```

Comment:
Since this logic is repeated, can probably extract this out to a method that accepts the type as a parameter.